fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2073,5 +2075,45 @@ public void formerIndexes() throws Exception {
             context.commit();
         }
     }
-    
+
+    @Test
+    public void boundaryPrimaryKeys() throws Exception {
+        final String indexName = "MySimpleRecord$num_value_unique";
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, TEST_SPLIT_HOOK);
+            recordStore.markIndexWriteOnly(indexName).get();
+            commit(context);
+        }
+
+        String bigOlString = Strings.repeat("x", SplitHelper.SPLIT_RECORD_SIZE + 2);
+        for (int i = -25; i < 25; i++) {
+            // Sparsify the primary keys so the records can be saved across boundaries.
+            saveAndSplitSimpleRecord(i * 39, bigOlString, i);
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context, TEST_SPLIT_HOOK);
+            Index index = recordStore.getRecordMetaData().getIndex(indexName);
+            try (OnlineIndexer indexer = OnlineIndexer.newBuilder()
+                    .setDatabase(fdb).setRecordStore(recordStore).setIndex(index)
+                    .build()) {
```

Comment:
Do we still need an `OnlineIndexer` in this test now that the method is in `FDBRecordStore`?